test/parallel/test-stream-map.js

```diff
@@ -24,7 +21,49 @@ const { setTimeout } = require('timers/promises');
     await Promise.resolve();
     return x + x;
   });
-  const result = [2, 4, 6, 8, 10];
+  (async () => {
+    assert.deepStrictEqual(await stream.toArray(), [2, 4, 6, 8, 10]);
+  })().then(common.mustCall());
```

Comment:
It might be sound to consider using a `.mjs` file for future tests like this so we can use TLA and skip all those async functions + then(mustCall).

Reply:
I was wondering if that was allowed but couldn’t find non-loader related .mjs tests. If it is I think it’s a good idea

Reply:
@aduh95 so can you confirm it's allowed :)? I'll port the file to `.mjs` I don't mind the tests running sequentially that'd actually make things easier to debug and the performance won't change since it's all microticks anyway.

Reply:
Nothing forbids it, so it's allowed (and there's no reason to avoid it since Node.js 10 is EOL and all maintained versions of Node.js support ESM). For example see `test/parallel/test-child-process-fork-url.mjs` and `test/parallel/test-fs-cp.mjs` which have little to do with loaders and have been written this way because it was more convenient.

Reply:
Ack I'll move all of these to mjs and simplify them as soon as the two PRs by outside contributors land to not create conflicts for them.